modules/loader/src/main/java/com/opengamma/strata/loader/csv/IborCapFloorTradeCsvPlugin.java

```diff
@@ -0,0 +1,330 @@
+import com.opengamma.strata.basics.index.IborIndex;
+import com.opengamma.strata.basics.index.Index;
+import com.opengamma.strata.basics.schedule.Frequency;
+import com.opengamma.strata.basics.schedule.PeriodicSchedule;
+import com.opengamma.strata.basics.schedule.StubConvention;
+import com.opengamma.strata.basics.value.ValueSchedule;
+import com.opengamma.strata.collect.io.CsvOutput.CsvRowOutputWithHeaders;
+import com.opengamma.strata.collect.io.CsvRow;
+import com.opengamma.strata.loader.LoaderUtils;
+import com.opengamma.strata.product.Trade;
+import com.opengamma.strata.product.TradeInfo;
+import com.opengamma.strata.product.capfloor.IborCapFloor;
+import com.opengamma.strata.product.capfloor.IborCapFloorLeg;
+import com.opengamma.strata.product.capfloor.IborCapFloorTrade;
+import com.opengamma.strata.product.common.PayReceive;
+import com.opengamma.strata.product.swap.IborRateCalculation;
+
+/**
+ * Handles the CSV file format for CapFloor trades.
+ */
+public class IborCapFloorTradeCsvPlugin implements TradeCsvParserPlugin, TradeTypeCsvWriter<IborCapFloorTrade> {
+
+  /**
+   * The singleton instance of the plugin.
+   */
+  public static final IborCapFloorTradeCsvPlugin INSTANCE = new IborCapFloorTradeCsvPlugin();
+
+  private static final String CAP_FLOOR_FIELD = "CapFloor";
+
+  private static final String STRIKE_FIELD = "Strike";
+
+  private static final ImmutableList<String> HEADERS = ImmutableList.<String>builder()
+      .add(CAP_FLOOR_FIELD)
+      .add(START_DATE_FIELD)
+      .add(END_DATE_FIELD)
+      .add(PAYMENT_FREQUENCY_FIELD)
+      .add(DIRECTION_FIELD)
+      .add(CURRENCY_FIELD)
+      .add(STRIKE_FIELD)
+      .add(NOTIONAL_FIELD)
+      .add(INDEX_FIELD)
+      .add(PREMIUM_AMOUNT_FIELD)
+      .add(PREMIUM_CURRENCY_FIELD)
+      .add(PREMIUM_DIRECTION_FIELD)
+      .add(PREMIUM_DATE_FIELD)
+      .add(DATE_ADJ_CNV_FIELD)
+      .add(DATE_ADJ_CAL_FIELD)
+      .add(START_DATE_CNV_FIELD)
+      .add(START_DATE_CAL_FIELD)
+      .add(END_DATE_CNV_FIELD)
+      .add(END_DATE_CAL_FIELD)
+      .add(STUB_CONVENTION_FIELD)
+      .add(ROLL_CONVENTION_FIELD)
+      .add(FIRST_REGULAR_START_DATE_FIELD)
+      .add(LAST_REGULAR_END_DATE_FIELD)
+      .add(OVERRIDE_START_DATE_FIELD)
+      .add(OVERRIDE_START_DATE_CNV_FIELD)
+      .add(OVERRIDE_START_DATE_CAL_FIELD)
+      .build();
+
+  //-------------------------------------------------------------------------
+  @Override
+  public Set<String> tradeTypeNames() {
+    return ImmutableSet.of("CAPFLOOR");
+  }
+
+  @Override
+  public Optional<Trade> parseTrade(
+      Class<?> requiredJavaType,
+      CsvRow baseRow,
+      List<CsvRow> additionalRows,
+      TradeInfo info,
+      TradeCsvInfoResolver resolver) {
+
+    if (requiredJavaType.isAssignableFrom(IborCapFloorTrade.class)) {
+      return Optional.of(resolver.parseCapFloorTrade(baseRow, info));
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public String getName() {
+    return "CapFloor";
```

Comment:
I think this should be `IborCapFloor`. The name is generally the trade class minus "Trade"